src/Components/TabItem.tsx

```diff
@@ -0,0 +1,21 @@
+import cn from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { Tab } from '../types/Tab';
+
+type Props = {
+  tab: Tab
+};
+
+export const TabItem: React.FC<Props> = ({ tab }) => {
+  const { id, title } = tab;
+  const { tabId } = useParams();
```

Comment:
You've already used `useParams` in the parent component. You can provide this ID via props.
Right now you will trigger `useParams()` hook on render every tab item, won't you?